CenterForOpenScience/osf.io
# [IN-302][OSF] Add `assets` attribute to provider endpoint.

## Purpose

Make asset information accessible through API.

## Changes

- Make a new `DevFileSystemStorage` which subclasses `FileSystemStorage` and use it for local dev development. 
- Add a new field to `ProviderSerializer` which serializes the assets information.
- Add simple tests.

## QA Notes

This should be tested together with [IN-294](https://openscience.atlassian.net/browse/IN-294). First upload provider assets through the admin panel. Then check the `providers` endpoint to make sure it returns the correct assets information.
```
{
    ...,
    attributes: {
        ...,
        assets: {
            <name>: <url>,
            ...,
        }
    }
}
```
## Side Effects

None. 

## Ticket

https://openscience.atlassian.net/browse/IN-302
